### ISSUE TITLE
test_queue_submit_many_order fails depending on which tests are run

### ISSUE THREAD

**Issue by mattwelborn** [MolSSI/QCFractal#395]
**Describe the bug**
`qcfractal/tests/test_storage.py::test_queue_submit_many_order` fails when a certain subset of tests are run. This suggests that it depends on the history of previous tests. 

**To Reproduce**
Code to reproduce the behavior:
`pytest -v -rsx --runslow qcfractal/ -k test_queue` always gives the error:
```
    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) insert or update on table "task_queue" violates foreign key constraint "task_queue_manager_fkey"
E       DETAIL:  Key (manager)=(test_manager) is not present in table "queue_manager".
E       
E       [SQL: UPDATE task_queue SET status=%(status)s, manager=%(manager)s, modified_on=%(modified_on)s WHERE task_queue.id IN (%(id_1)s)]
E       [parameters: {'status': 'running', 'manager': 'test_manager', 'modified_on': datetime.datetime(2019, 9, 3, 13, 45, 47, 596899), 'id_1': 3}]
E       (Background on this error at: http://sqlalche.me/e/gkpj)

../../../anaconda3/envs/test/lib/python3.7/site-packages/sqlalchemy/engine/default.py:552: IntegrityError
```
`pytest -v -rsx --runslow qcfractal/` passes.
`pytest -v -rsx --runslow qcfractal/ -k test_queue_submit_many_order` passes or fails, but I don't know under what conditions.

**Expected behavior**
`test_queue_submit_many_order` should pass regardless of which test were run before it.

**Comment by dgasmith** [comment]
@doaa-altarawy Can you take a look at this one? This likely goes back to some sort of async timing issue.

**Comment by dgasmith** [comment]
Found it, pushing a fix to #385.
